subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -387,35 +421,24 @@ static uint8_t read_feature_cb(struct bt_conn *conn, uint8_t err,
 
 int bt_otc_register(struct bt_otc_instance_t *otc_inst)
 {
-	static bool l2cap_registered;
-
 	for (int i = 0; i < ARRAY_SIZE(otc_insts); i++) {
+		int err;
 
 		if (otc_insts[i].otc_inst) {
 			continue;
 		}
 
 		BT_DBG("%u", i);
-
-		if (!l2cap_registered) {
-			int err = bt_l2cap_server_register(&ots_l2cap_coc);
-
-			if (err < 0 && err != -EADDRINUSE) {
-				BT_ERR("Unable to register OTS_psm (0x%4x)",
-				ots_l2cap_coc.psm);
-				return -ENOEXEC;
-			}
-
-			BT_DBG("L2CAP psm 0x%4x sec_level %u registered",
-			ots_l2cap_coc.psm, ots_l2cap_coc.sec_level);
-
-			l2cap_registered = true;
+		err = bt_gatt_ots_l2cap_register(&otc_insts[i].l2cap_ctx);
+		if (err) {
+			BT_WARN("Could not register L2CAP context %d", err);
+			return err;
 		}
 
 		otc_insts[i].otc_inst = otc_inst;
-
 		return 0;
 	}
+
 	return -ENOMEM;
 }
 
```

Comment:
How does one use the `bt_otc_unregister`? The instance index is not returned from the `bt_otc_register`, maybe this should also take a `struct bt_otc_instance_t *otc_inst` instead of `index` as a parameter?

Also should the l2cap context also be unregistered?

Though, this function is not used by anyone so maybe it can be resolved in a follow up PR.

Reply:
https://github.com/zephyrproject-rtos/zephyr/issues/42649